src/threadinterrupt.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <mutex>
+#include <condition_variable>
+#include <atomic>
+
+class CThreadInterrupt
+{
+public:
+    CThreadInterrupt(std::condition_variable& condIn, std::mutex& mutIn) : cond(condIn), mut(mutIn), val(false) {}
+    void reset()
```

Comment:
Please no new implementations-in-headers. I know its mostly single-line stuff but we're fighting to reduce compile-time memory usage, so every little bit helps. :)